### PR TITLE
Porting ladder diagrams to node.js

### PR DESCRIPTION
**Pull request by martinthomson**
Running this stuff through the webkit command line on a mac is fine, if you have that.  Node is 10% more awesome.

### CODE REVIEW

File: ladder/debug.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+var DEBUG = false;
+var debug = function(msg) {
+        if (DEBUG) {
+            console.warn("<!-- " + msg + " -->");
+        }
```
Comment (ekr) [inline ladder/debug.js:7]: Why is this a comment?

File: ladder/debug.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    };
+
+debug.die = function(msg) {
+    console.error("ERROR: " + msg);
+    throw msg;
```
Comment (ekr) [inline ladder/debug.js]: Can you write a unit test that this works?

File: ladder/ladder-parser.js [hunk 3]
```diff
@@ -1,173 +1,171 @@
...
-};
-
-var DEBUG = false;
-
-var debug = function(msg) {
-    if (DEBUG)
-        print("DEBUG: " + msg);
-};
+var debug = require('./debug');
+var Ladder = require('./ladder-diagram');
```
Comment (ekr) [inline ladder/ladder-parser.js:18]: What does the ./ do here if you are not int he local directory?

Comment (martinthomson) [inline ladder/ladder-parser.js:18]: It is evaluated relative to the current file.  Like __directory.

File: ladder/debug.js (hunk 2 above)
Comment (martinthomson) [inline ladder/debug.js]: This works, but I'll update with process.exit(1).

File: ladder/debug.js (hunk 1 above)
Comment (martinthomson) [inline ladder/debug.js:7]: 2>&1

**Pull request merged**